GCC10: Fix issues related to common symbols

1480c1 opened the pull request:
Closes #999

The issue is caused by the fact that `aom_dsp_rtcd.h` includes `common_dsp_rtcd.h`, and use the same define to trigger using `extern`, causing `aom_dsp_rtcd.o` to have the same function pointers as `common_dsp_rtcd.o`

tianjunwork commented on Source/Lib/Decoder/Codec/EbDecProcess.c:
```diff
@@ -135,6 +135,13 @@ EbErrorType dec_dummy_creator(EbPtr *object_dbl_ptr, EbPtr object_init_data_ptr)
+    static uint8_t num_groups = 0;
+#elif defined(__linux__)
+    static cpu_set_t group_affinity;
+#endif
```
where are they used in this file?

1480c1 replied:
These are used in the expansion of the `EB_CREATE_THREAD` macro which is expanded from the `EB_CREATE_THREAD_ARRAY` macro

tianjunwork commented on Source/Lib/Encoder/Codec/aom_dsp_rtcd.c:
```diff
@@ -14,7 +14,7 @@
 */
 
-#define RTCD_C
+#define AOM_RTCD_C
```
Good idea!

tianjunwork commented on Source/Lib/Encoder/Codec/aom_dsp_rtcd.h:
```diff
@@ -559,7 +558,6 @@ extern "C" {
     RTCD_EXTERN uint32_t(*nxm_sad_kernel_sub_sampled)(const uint8_t *src, uint32_t src_stride, const uint8_t *ref, uint32_t ref_stride, uint32_t height, uint32_t width);
     RTCD_EXTERN uint32_t(*nxm_sad_kernel)(const uint8_t *src, uint32_t src_stride, const uint8_t *ref, uint32_t ref_stride, uint32_t height, uint32_t width);
     RTCD_EXTERN uint32_t(*nxm_sad_avg_kernel)(uint8_t *src, uint32_t src_stride, uint8_t *ref1, uint32_t ref1_stride, uint8_t *ref2, uint32_t ref2_stride, uint32_t height, uint32_t width);
-    RTCD_EXTERN void(*avc_style_luma_interpolation_filter)(EbByte ref_pic, uint32_t src_stride, EbByte dst, uint32_t dst_stride, uint32_t pu_width, uint32_t pu_height, EbByte temp_buf, EbBool skip, uint32_t frac_pos, uint8_t fractional_position);
```
Thanks for cleaning up the dup definitions.

tianjunwork commented on Source/Lib/Encoder/Globals/EbEncHandle.c:
```diff
@@ -309,7 +308,7 @@ void init_intra_dc_predictors_c_internal(void);
 void eb_av1_init_me_luts(void);
 
-void switch_to_real_time(){
+static void enc_switch_to_real_time(){
```
since it is static now, is enc_/dec_ needed?

1480c1 replied:
Not really, but just to make sure there isn't confusion on which `switch_to_real_time` a symbol might be

tianjunwork commented on Source/Lib/Encoder/Globals/EbEncHandle.c:
```diff
@@ -97,19 +97,18 @@
  **************************************/
-
-uint8_t                          num_groups = 0;
+static uint8_t num_groups = 0;
```
the style needs to be kept the same.

1480c1 replied:
Done

tianjunwork commented:
Also, could you add a new line in https://github.com/OpenVisualCloud/SVT-AV1/blob/86df140972c6140f5efa1042ad573a3aeaa7b41b/Source/Lib/Common/Codec/common_dsp_rtcd.h#L181?

1480c1 commented:
Dropped the travis-ci job and moved it to github actions. Although it's labeled as `gcc-10`, it doesn't use gcc 10, but rather emulates the braking change of adding `-fno-common` to the CFLAGS

tianjunwork commented:
Hi @OmranAbdelkafi , could you review this one first? GCC 10 is already adopted by appveyor and causes CI failure with the recent PRs. e.g. https://ci.appveyor.com/project/tianjunwork/svt-av1/builds/32790702/job/g4bhsi13b1csyopo

EricLamVCD commented:
Tested obj-1 8bit and 10bit clips: 
  | Speed Deviation  | BD-Rate   Deviation
-- | -- | --
M0 - 8bit | 0.23% | 0.00%
M4 - 8bit | 0.06% | 0.00%
M8 - 8bit | 0.08% | 0.00%
M0 - 10bit | 0.13% | 0.00%
M4 - 10bit | -0.13% | 0.00%
M8 - 10bit | -0.07% | 0.00%
